horace_core/sqw/@Experiment/Experiment.m

```diff
@@ -4,9 +4,9 @@
     properties(Access=private)
         % if no other instrument input is provided Exeriment instrument
         % remain empty.
-        instruments_ = unique_references_container('GLOBAL_NAME_INSTRUMENTS_CONTAINER','IX_inst');
-        detector_arrays_ = []
-        samples_ = unique_references_container('GLOBAL_NAME_SAMPLES_CONTAINER','IX_samp');
+        instruments_ = "initialised in constructor";
+        detector_arrays_ = "initialised in constructor";
+        samples_ = "initialised in constructor";
```

Comment:
empty array would be a legal value, this clarifies intent and provides an illegal value to be overwritten

Reply:
I would say assigning these values is very bad practice, forcing you to do some additional movements to avoid this. 
Empty experiment is a valid class, while all this palaver does not add any clarity.

Reply:
Please clarify what additional movements are required. Note that empty Experiment remains a valid class. Pretty much the first thing the constructor does is to assign these as unique_references_containers, so an empty Experiment will be constructed correctly due to that. The problem is that making that assignment here does not work; it must be done in the constructor. By forcing the issue here, developers are made aware of this, and any attempt to bypass that when recoding the constructor will then encounter these invalid values which will reinforce the point.